### PR TITLE
[bugfix] Add libpath option to bootstrap script

### LINKED ISSUE #1492: bootstrap script cannot handle custom python install paths
The bootstrap is not able to deal with custom python3 install locations. It requires that `pip` is found with the following folder structure:
```
external/usr/lib/python$pyver/site-packages
```
I am referring to this [line](https://github.com/eth-cscs/reframe/blob/d08a2cb66703efd7e860633ade28244713d43a92/bootstrap.sh#L54). This is fine for system python, but when I use a python3 installed on a different location, I get the following:

```
==> 941182: reframe: Executing phase: 'install'                                    
==> [2020-09-21-09:03:36.432361, 941186] '/usr/bin/bash' './bootstrap.sh' ''       
==> python3 -m ensurepip --root external/ --default-pip                            
Looking in links: /tmp/tmpsiva_b_g                                                 
Requirement already satisfied: setuptools in /my/spack/install/path/opt/spack/linux-centos8-broadwell/gcc-8.3.1/py-setuptools-46.1.3-n2by5yrsutgd3v6gwpbeelenw3eox2z5/lib/python3.7/site-packages (46.1.3)
Collecting pip                                                                     
Installing collected packages: pip                                                 
Successfully installed pip-19.2.3                                                  
==> python3 -m pip install -q --upgrade pip --target=external/                     
/my/spack/install/path/opt/spack/linux-centos8-broadwell/gcc-8.3.1/python-3.7.7-w27qutd7t3uor2sqlpdsfyv4fbfdiole/bin/python3: No module named pip 
==> python3 -m pip install -q -r requirements.txt --target=external/ --upgrade  
/my/spack/install/path/opt/spack/linux-centos8-broadwell/gcc-8.3.1/python-3.7.7-w27qutd7t3uor2sqlpdsfyv4fbfdiole/bin/python3: No module named pip 
==> [2020-09-21-09:03:39.464072, 941186] Installing /tmp/hvictor/spack-stage/spack-stage-reframe-3.1-kxtmks6dzyny5y6mssaikwry6sgxlu5w/spack-src to /my/spack/install/path/opt/spack/linux-centos8-broadwell/gcc-8.3.1/reframe-3.1-kxtmks6dzyny5y6mssaikwry6sgxlu5w
```

The actual place where `pip` got installed was
```
/my/spack/install/path/opt/spack/linux-centos8-broadwell/gcc-8.3.1/reframe-3.1-kxtmks6dzyny5y6mssaikwry6sgxlu5w/external/my/spack/install/path/opt/spack/linux-centos8-broadwell/gcc-8.3.1/python-3.7.7-w27qutd7t3uor2sqlpdsfyv4fbfdiole
```

### PR DESCRIPTION
**Pull request by victorusu**
This adds the `-p` option to pass a path to enable a different python path.

Fixes #1492.

### CODE REVIEW

File: bootstrap.sh [hunk 1]
```diff
@@ -57,7 +63,8 @@ fi
 # everything under `external/`. That's why include both in the PYTHONPATH
 
 export PATH=$(pwd)/external/usr/bin:$PATH
-export PYTHONPATH=$(pwd)/external:$(pwd)/external/usr/lib/python$pyver/site-packages:$PYTHONPATH
+export PYTHONPATH=$(pwd)/external:$(pwd)/external/${libpath}/python$pyver/site-packages:$PYTHONPATH
+echo "Setting PYTHONPATH to ${PYTHONPATH}"
```
Comment (vkarak) [inline bootstrap.sh:67]: Remove this line.

File: bootstrap.sh [hunk 2]
```diff
@@ -57,7 +63,8 @@ fi
 # everything under `external/`. That's why include both in the PYTHONPATH
 
 export PATH=$(pwd)/external/usr/bin:$PATH
-export PYTHONPATH=$(pwd)/external:$(pwd)/external/usr/lib/python$pyver/site-packages:$PYTHONPATH
+export PYTHONPATH=$(pwd)/external:$(pwd)/external/${libpath}/python$pyver/site-packages:$PYTHONPATH
```
Comment (vkarak) [inline bootstrap.sh:66]: I guess you expect `libpath` to be an absolute path, so I would write this as

```suggestion
export PYTHONPATH=$(pwd)/external:$(pwd)/external${libpath}/python$pyver/site-packages:$PYTHONPATH
```

File: bootstrap.sh [hunk 3]
```diff
@@ -24,20 +24,26 @@ usage()
     echo "Usage: $0 [-h] [+docs]"
     echo "Bootstrap ReFrame by pulling all its dependencies"
     echo "  -P EXEC  Use EXEC as Python interpreter"
+    echo "  -p PATH  Use PATH as Python lib. Default /usr/lib"
```
Comment (vkarak) [inline bootstrap.sh:27]:
```suggestion
    echo "  -L PATH  Prefix of Python libraries (default: '/usr/lib')"
```

**Review by vkarak — CHANGES_REQUESTED** [review]

File: bootstrap.sh [hunk 4]
```diff
@@ -63,8 +63,7 @@ fi
 # everything under `external/`. That's why include both in the PYTHONPATH
 
 export PATH=$(pwd)/external/usr/bin:$PATH
-export PYTHONPATH=$(pwd)/external:$(pwd)/external/${libpath}/python$pyver/site-packages:$PYTHONPATH
-echo "Setting PYTHONPATH to ${PYTHONPATH}"
+export PYTHONPATH=$(pwd)/external:$(pwd)/external${libpath}/python$pyver/site-packages:$PYTHONPATH
```
Comment (vkarak) [inline bootstrap.sh:66]: I think we should also a small check that the `libpath` is an absolute path and issue an error if not, or prefix it with `pwd`.

**Review by vkarak — CHANGES_REQUESTED** [review]

**Comment by victorusu** [comment]
Further investigation showed that the problem displayed in #1492 is not related to the external dependencies, but to the way pip gets installed on the system.
In this case, I think that a better solution is to use the system- or package manager- provided version of pip.

**Pull request closed without merging**